### PR TITLE
disable Compile button on loading

### PR DESCRIPTION
**Pull request by LianaHus**

### CODE REVIEW

File: src/app/tabs/compileTab/compilerContainer.js [hunk 1]
```diff
@@ -37,6 +37,7 @@ class CompilerContainer {
     }
     if (!this._view.compilationButton) return
     const button = this.compilationButton(name.split('/').pop())
+    this._disableCompileBtn(!name || name === '')
```
Comment (GrandSchtroumpf) [inline src/app/tabs/compileTab/compilerContainer.js]: if `name === '' // true` then `!name // true`. So you don't need `|| name === ''`.
In Javascript : 
`!0 // true`
`!"" // true`
`!null // true`
`!undefined // true`
BUT
`![] // false`
`!{} // false`

File: src/app/tabs/compileTab/compilerContainer.js [hunk 2]
```diff
@@ -103,19 +106,31 @@ class CompilerContainer {
    */
   compilationButton (name) {
```
Comment (GrandSchtroumpf) [inline src/app/tabs/compileTab/compilerContainer.js]: You can add a default value directly inside the parameter : 
```javascript
comilationButton (name = '') {
  const displayed = name || '<no file selected>'
  const disabled = name ? '' : 'disabled'
  ...
}
```

File: src/app/tabs/compileTab/compilerContainer.js [hunk 3]
```diff
@@ -103,19 +106,31 @@ class CompilerContainer {
...
+    const compileBtn = yo`
+      <button id="compileBtn" class="btn btn-primary btn-block ${disabled}" title="Compile" onclick="${this.compile.bind(this)}">
         <span>${this._view.compileIcon} Compile ${displayed}</span>
       </button>
-    </div>`
-    if (name === '') {
-      el.setAttribute('disabled', 'true')
-    }
+    `
+    let el = yo`
```
Comment (GrandSchtroumpf) [inline src/app/tabs/compileTab/compilerContainer.js]: To make it more "simple" you can also do : 
```javascript
return yo`<div class="px-2 mt-2 pb-0 d-flex">${compileBtn}</div>`
```

File: src/app/tabs/compileTab/compilerContainer.js [hunk 4]
```diff
@@ -103,19 +106,31 @@ class CompilerContainer {
...
+      <div class="px-2 mt-2 pb-0 d-flex">
+        ${compileBtn}
+      </div>
+    `
     return el
   }
 
+  _disableCompileBtn (disable) {
```
Comment (GrandSchtroumpf) [inline src/app/tabs/compileTab/compilerContainer.js]: Can you add a comment on the method ? Also as the param is a boolean, I would recommend `isDisabled` or `shouldDisable`
```javascript
/**
 * Toggle the compile button
 * @param {boolean} isDisabled Should disable or enable
 */
```

File: src/app/tabs/compileTab/compilerContainer.js [hunk 5]
```diff
@@ -103,19 +106,31 @@ class CompilerContainer {
...
+    let btn = document.querySelector("button[id='compileBtn']")
```
Comment (GrandSchtroumpf) [inline src/app/tabs/compileTab/compilerContainer.js]: Id are supposed to be unique, better to do: `document.getElementById('compileBtn')` (it's much faster)

File: src/app/tabs/compileTab/compilerContainer.js (hunk 4 above)
Comment (LianaHus) [reply src/app/tabs/compileTab/compilerContainer.js]: I don't think it needs comment, the name is clear enough.

**Pull request merged**